fakestorage/upload.go

```diff
@@ -354,7 +342,8 @@ func (s *Server) multipartUpload(bucketName string, r *http.Request) jsonRespons
 		},
 		Content: notImplementedSeeker{io.NopCloser(io.MultiReader(partReaders...))},
 	}
-	obj, err = s.createObject(obj)
+
+	obj, err = s.createObject(obj, *conditions)
```

Comment:
with the change above:

```suggestion
	obj, err = s.createObject(obj, conditions)
```